From 4e69793b9b77a394b8f7842a78de235eaf3859df Mon Sep 17 00:00:00 2001
From: andrew-2e128 <andrew.2e128@outlook.com>
Date: Sun, 12 Apr 2026 01:51:05 +0200
Subject: [PATCH] nvidia: Update params in the infinite loop

`params` was getting loaded only once at startup, making the initial setup determine what stats will be loaded ignoring preset change
---
 src/nvidia.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/nvidia.cpp b/src/nvidia.cpp
index f9e2837fdf..1c472dda20 100644
--- a/src/nvidia.cpp
+++ b/src/nvidia.cpp
@@ -224,9 +224,10 @@ void NVIDIA::get_instant_metrics_xnvctrl(struct gpu_metrics *metrics) {
 void NVIDIA::get_samples_and_copy() {
     struct gpu_metrics metrics_buffer[METRICS_SAMPLE_COUNT] {};
     auto logger_ref = logger; // inc ref count, to avoid destruction of logger.
-    auto params = get_params();
-    auto params_p = params.get(); // avoid destruction while we are getting samples...
     while(!stop_thread) {
+        auto params = get_params();
+        auto params_p = params.get(); // avoid destruction while we are getting samples...
+
 #ifndef TEST_ONLY
         if (HUDElements.g_gamescopePid > 0 && HUDElements.g_gamescopePid != pid) {
             pid = HUDElements.g_gamescopePid;
